### PR TITLE
Remove whatwg-fetch

### PR DESCRIPTION
**Pull request by calebmer**
As per an [earlier discussion](https://github.com/apollostack/apollo-client/pull/1122#issuecomment-270565539), this PR removes `whatwg-fetch` and replaces it with a runtime warning if a global `fetch` implementation was not found :+1:

Looks like this one kills another 15% in file size taking us down to 21.7 kB gzipped.

- [x] Make sure all of the significant new logic is covered by tests
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
+import './fetch';
```
Comment (stubailo) [inline src/index.ts]: This should be imported in the network interface only - that's the only part that depends on fetch.

Comment (calebmer) [inline src/index.ts]: The `whatwg-fetch` import was in both `transport/networkInterface.js` and `transport/batchedNetworkInterface.js`. Both of which are also imported into this file, so the same behavior would occur if we moved this to one or the other files.

With that in mind, would you still like it moved to the network interface, and if so which one? Or both?

Comment (stubailo) [inline src/index.ts]: Oh, that's a good point. We don't have any facility for not loading part of the code.

Given that, can we print the warning only when `createNetworkInterface` is called?

**Comment by helfer** [comment]
@calebmer Looks great!! Can you create a PR on the docs to make sure we don't forget to tell people that they have to install a fetch implementation now?

**Comment by calebmer** [comment]
@helfer :+1:

- https://github.com/apollostack/angular2-docs/pull/28
- Accidentally commit directly to the React docs, let me know if you want any changes: https://github.com/apollostack/react-docs/commit/294db7e30968b800462564f488b879a126dfefd5

**Pull request merged**